src/operator/mkldnn/mkldnn_pooling-inl.h

```diff
@@ -0,0 +1,220 @@
+      bwd_in_diff_->get_output_ptr(ingrad.dptr_);
+    } else {
+      PoolingOp<cpu, Reducer, DType>::Backward(ctx, out_grad, in_data, out_data,
+                                               req, in_grad, aux_args);
+    }
+  }
+
+ private:
+  inline void Init(mshadow::Stream<cpu> *s, const std::vector<TBlob> &in_data,
+                   const std::vector<TBlob> &out_data) {
+    using namespace mshadow;
+    CHECK_EQ(in_data.size(), 1);
+    CHECK_EQ(out_data.size(), 1);
+    if (!init_mkldnn_) {
+      init_mkldnn_ = true;
+      if (this->param_.kernel.ndim() == 2) {
+        Tensor<cpu, 4, DType> data =
+            in_data[pool_enum::kData].get<cpu, 4, DType>(s);
+        Tensor<cpu, 4, DType> out =
+            out_data[pool_enum::kOut].get<cpu, 4, DType>(s);
+
+        size_t src_sizes[4], src_strides[4];
+        size_t dst_sizes[4], dst_strides[4];
+        size_t kernel_size[2];
+        size_t kernel_stride[4];
+        int src_offset[2];
+        size_t dim = 4;
+
+        const size_t batch_size = data.size(0);
+        const size_t input_channels = data.size(1);
+        const size_t input_height = data.size(2);
+        const size_t input_width = data.size(3);
+        const size_t output_channels = input_channels;
+        const size_t output_height = out.size(2);
+        const size_t output_width = out.size(3);
+
+        src_sizes[0] = input_width;
+        src_sizes[1] = input_height;
+        src_sizes[2] = input_channels;
+        src_sizes[3] = batch_size;
+        src_strides[0] = 1;
+        src_strides[1] = src_sizes[0];
+        src_strides[2] = src_sizes[0] * src_sizes[1];
+        src_strides[3] = src_sizes[0] * src_sizes[1] * src_sizes[2];
+        dst_sizes[0] = output_width;
+        dst_sizes[1] = output_height;
+        dst_sizes[2] = output_channels;
+        dst_sizes[3] = batch_size;
+        dst_strides[0] = 1;
+        dst_strides[1] = dst_sizes[0];
+        dst_strides[2] = dst_sizes[0] * dst_sizes[1];
+        dst_strides[3] = dst_sizes[0] * dst_sizes[1] * dst_sizes[2];
+        kernel_size[0] = this->param_.kernel[0];
+        kernel_size[1] = this->param_.kernel[1];
+        kernel_stride[0] = this->param_.stride[0];
+        kernel_stride[1] = this->param_.stride[1];
+        src_offset[0] = -this->param_.pad[0];
+        src_offset[1] = -this->param_.pad[1];
+
+        size_t full_model_output_height =
+            1 + static_cast<int>(ceil(static_cast<float>(
+                    input_height + 2 * this->param_.pad[0] -
+                    this->param_.kernel[0]) / this->param_.stride[0]));
+        size_t full_model_output_width =
+            1 + static_cast<int>(ceil(static_cast<float>(
+                    input_width + 2 * this->param_.pad[0] -
+                    this->param_.kernel[0]) / this->param_.stride[0]));
+        if ((full_model_output_width == output_width) &&
+            (full_model_output_height == output_height)) {
+          support_mkldnn_ = true;
```

Comment:
No. 
As our experiments, MKLDNN only support rounds up pooling. 
But sometimes, rounds up pooling and rounds down pooling has same output size.
That piece of codes determines whether the output size is same as rounds up pooling.
I know it's confusing so update the pooling code. Hope it can be better.